src/main/java/com/artipie/Pie.java

```diff
@@ -92,15 +96,37 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
         }
         final String repo = parts[0];
         Logger.debug(this, "Slice repo=%s", repo);
+        final MutableBoolean found = new MutableBoolean(true);
         return new AsyncSlice(
             CompletableFuture.supplyAsync(
                 () -> new Unchecked<>(this.settings::storage).value()
+            ).thenComposeAsync(
+                storage -> storage.value(new Key.From(String.format("%s.yaml", repo)))
+            ).exceptionally(
+                throwable -> {
```

Comment:
A bad approach. Resulted content should inherit the error.

Reply:
@Sammers21 I agree, but didn't find a way to do it. What do you suggest?

Reply:
@Sammers21 any insight?

Reply:
@Sammers21 ping

Reply:
Sorry for the delay. You can use RxStorage or RxJava 2 API: 

```java
Publisher<ByteBuffer> pub = new RxStorageWrapper(storage).value(key).flatMapPublisher(pub -> pub)
```

Reply:
@Sammers21 And this way, the resulted content would inherit the error? How can we detect it to return 404? When you say "Bad approach", you are suggesting the boolean, right?

Reply:
Using this approach the resulted publisher will inherit the error.

Reply:
@Sammers21 Got it. Thanks. Please see my last commit.

Reply:
@HDouss , the publisher will emit an error. You need to subscribe in order to receive that signal.

Reply:
@Sammers21 Who would subscribe? Please keep in mind that we are chaining completableFutures that should return a CompletionStage<Slice> at last and passed it to AsyncSlice. Reading the content with blockingGet isn't enough? That is throwing an exception (tested it locally).